### ISSUE TITLE
Issues using pysam in an AWS EC2 instance

### ISSUE THREAD

**Issue by vb-tony** [pysam-developers/pysam#1079]
Hello,

I have set up an EC2 instance in AWS with the Ubuntu AMI and am trying to use pysam in it. I have noticed that the `pip install pysam` command does not seem to take into account any of the environment variables. I tested this by:

`export HTSLIB_CONFIGURE_OPTIONS=--disable-libcurl`
`pip install pysam`

But when attempting to run the following: `tbxFile = pysam.TabixFile("https://path/to/a/file", index="https://path/to/a/file"+".tbi", mode="r")`, it doesn't say that the file was not found but rather recognizes that it's an HTTPS protocol file. I am testing this because I am trying to enable S3 paths for my pysam install.

Has anyone had any experience with installing pysam on an EC2 machine?

**Comment by vb-tony** [comment]
Just found out there is a Google group for questions. Will close this.

**Comment by vb-tony** [comment]
Reopening because the Google group doesn't seem very active.

**Comment by vb-tony** [comment]
Figured it out! I was able to resolve the issue by utilizing a virtual environment. Using virtualenv with Python and building pysam from the virtual environment, I was able to resolve my weird dependency issues. This was done on a Ubuntu platform.

Steps:
1. Export desired HTSLIB_CONFIGURE_OPTIONS
2. Install libraries per HTSLIB install instructions
2. Set up virtualenv: `virtualenv --python=python3 env`
3. Instally Cython: `pip install cython`
4. Clone the pysam repo 
5. `cd` into repo and run `python3 setup.py install`

Leaving this comment as a reference for anyone else running into this problem.
